### DISCUSSION TITLE
Disabling optimisations for limited subprojects

### DISCUSSION THREAD

**Discussion by mitchgrout** [mesonbuild/meson#12902] [Q&A]
As I understand, `meson setup -Doptimization=3` is a global option, so any subprojects I may have will also be compiled at `-O3`. I would like to disable this for certain subprojects; what would be the cleanest way of doing this?

One option is to add a `<lang>_args=['-O3']` to my build targets, but this would only compile my project at `-O3`, and leave all my subprojects at the default optimization level.

**Comment by vvainola** [comment]
Would be nice to have this other way around as well that subprojects have optimizations enabled and main project is without optimizations. Currently adding `-Dsubproject:optimization=3` in setup prints it out as changed but it does nothing. Also in VS backend using override_options to change optimization level does not work either.
